src/main/java/task/TaskList.java

```diff
@@ -0,0 +1,120 @@
+package task;
+
+import java.util.ArrayList;
+
+/**
+ * List of user-specified tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructs TaskList object with empty list.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+
+    /**
+     * Constructs TaskList object with existing list of tasks.
+     * @param tasks List of task.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Removes task from the list according to its index.
+     * @param index Task index.
+     * @return Removed task.
+     */
+    public Task remove(int index) {
+        Task task = tasks.get(index);
+        tasks.remove(index);
+        return task;
+    }
+
+    /**
+     * Marks specific task in the list as done according to its index.
+     * @param index Task index.
+     * @return Done task.
+     */
+    public Task done(int index) {
+        Task task = tasks.get(index);
+        task.markDone();
+        return task;
+    }
+
+    /**
+     * Adds task into the list.
+     * @param task Specified task.
+     */
+    public void add(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Lists all the tasks in the list.
+     */
+    public String list() {
+        int counter = 1;
+        StringBuilder str = new StringBuilder();
+
+        //list out all the texts from the user
+        for (Task t: tasks) {
+            str.append(counter).append(".").append(t.toString()).append("\n");
+            counter++;
+        }
+
+        return str.toString();
+    }
+
+    /**
+     * Lists all the tasks in the list provided by the user.
+     * @param taskList List of tasks.
+     */
+    public String list(ArrayList<Task> taskList) {
+        int counter = 1;
+        StringBuilder str = new StringBuilder();
+
+        //list according to given list
+        for (Task t: taskList) {
+            str.append(counter).append(".").append(t.toString()).append("\n");
```

Comment:
Good use of StringBuilder! I wanted to suggest using String.format but I realised that StringBuilder is actually faster! However a good thing about String.format is that it is more readable. More info can be found here: https://stackoverflow.com/questions/44117788/performance-between-string-format-and-stringbuilder/44127617

Reply:
Yes, I did consider String.format before, but I felt StringBuilder is more neat for me! I guess it's just a matter of preference! :)